pkg/scheduler/plugins/topology/job_filtering.go

```diff
@@ -52,12 +53,24 @@ func (t *topologyPlugin) subSetNodesFn(
 		return []node_info.NodeSet{}, nil
 	}
 
+	// Sorting the tree for both packing and closest preferred level domain scoring
+	preferredLevel := DomainLevel(subGroup.GetTopologyConstraint().PreferredLevel)
+	requiredLevel := DomainLevel(subGroup.GetTopologyConstraint().RequiredLevel)
+	maxDepthLevel := preferredLevel
+	if maxDepthLevel == "" {
+		maxDepthLevel = requiredLevel
+	}
+	sortTree(topologyTree.DomainsByLevel[rootLevel][rootDomainId], maxDepthLevel)
+	if preferredLevel != "" {
+		t.subGroupNodeScores[subGroup.GetName()] = calculateNodeScores(topologyTree.DomainsByLevel[rootLevel][rootDomainId], preferredLevel)
```

Comment:
This will run on all nodes before we have a chance to maybe filter out some of them. Maybe it can run some how lazily? If it is too complicated we can do it in another PR later.

Reply:
I can't think of such a way at the moment. Do you have something in mind?